torchtune/datasets/multimodal/_llava_instruct.py

```diff
@@ -89,6 +89,8 @@ def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
                     role="system", content=self.new_system_prompt, masked=True, eot=True
                 )
             )
+        
+        # Add in image stuffs / load from file
```

Comment:
I am leaning towards handling image loading in the model transform and keeping a URL here, so we don't need to pass in a heavy PIL image everywhere Messages is used. We only load in the image when it's absolutely needed. For tokenize_messages, messages would be more lightweight since it won't contain the actual image.

Reply:
Re passing around PIL images, do you mean that there are implications for memory to doing something like this? Cause if we are just passing stuff by reference it shouldn't be an issue, right? 

If there are a bunch of operations that we want to apply to messages that'd be more convenient to have the file path for, that'd be one thing. But it seems to me like most (nontrivial) functions we're gonna call on Messages containing images would be those that act on the image and not the path. 

In that case I'd be inclined to have things as a PIL because it's the raw datatype of an image. It's analogous to the raw string we have in the content field of a Message with text type. I think {"text" -> str, "image" -> PIL} is a lot more natural for a user than {"text" -> str, "image" -> Path}. But lmk if I'm missing the point here

Reply:
See comment in FAQs. Generally, I agree, but I'm making the tradeoff for "weight" of dependencies.

Reply:
> Right now, it's possible to train and run inference without downloading extra things needed for multimodal like PIL and torchvision

I think Joe's point here is quite valid. I agree having it PIL image by default is more intuitive but as of now multimodal is not prevalent enough as a use case.

Reply:
OK yeah that's fine with me